pyogrio/geopandas.py

```diff
@@ -4,8 +4,26 @@
 from pyogrio.raw import read, write
 
 
+def _stringify_path(path):
+    """
+    Convert path-like to a string if possible, pass-through other objects
+    """
+    if isinstance(path, str):
+        return path
+
+    # checking whether path implements the filesystem protocol
+    try:
```

Comment:
Not sure if it is more efficient, but I've always been a bit uncomfortable using exceptions to test attributes.

What about 

```Python
if hasattr(path, "__fspath__()"):
    return path.__fspath__()
```